lib/token.js

```diff
@@ -0,0 +1,282 @@
+'use strict';
+
+const types = require('./types');
+const util = require('util');
+
+const _Murmur3TokenType = types.dataTypes.getByName('bigint');
+const _RandomTokenType = types.dataTypes.getByName('varint');
+const _OrderedTokenType = types.dataTypes.getByName('blob');
+
+/**
+ * Represents a token on the Cassandra ring.
+ */
+class Token {
+  constructor(value) {
+    this._value = value;
+  }
+
+  /**
+   * @returns {{code: number, info: *|Object}} The type info for the
+   *                                           type of the value of the token.
+   */
+  getType() {
+    throw new Error('You must implement a getType function for this Token instance');
+  }
+
+  /**
+   * @returns {*} The raw value of the token.
+   */
+  getValue() {
+    return this._value;
+  }
+
+  /**
+   * Encodes the underlying value into a buffer that can be
+   * used for routing.
+   * 
+   * @param {Encoder} encoder Encode for encoding to buffer.
+   * @returns {Buffer} buffer data for the token.
+   */
+  toBuffer(encoder) {
```

Comment:
Yeah the initial idea was to provide it as a convenience function for those looking to convert Tokens into Buffers so they may be fed into `getReplicas`, but since then I've allowed `getReplicas` to take `Token` as an input, so it's no longer needed, so I think we can remove it (will do).